lib/elastic_record/index.rb

```diff
@@ -80,7 +80,7 @@ def real_connection
 
     def get(end_path, doctype = nil, json = nil)
       path = "/#{alias_name}"
-      path += "/#{doctype.name}" if doctype
+      path += "/#{ElasticRecord::Config.default_doctype_name}" if doctype
```

Comment:
This doesn't seem to match with the `if` at the end or the argument passed in. Is the argument left here (and other places) for api compatibility? If so, when does it get removed?

Reply:
Yes, it is for api compatibility.
As for when it gets removed, that's a good question. These changes force all models to be mapped in an index with a single doctype (`doc` by default). Because of this, it's no longer valid to use a dynamic doctype the ES rest API. The complication is that `get` isn't necessarily always used for queries. So there isn't a clean way to inject the given doctype name into the `json` body to force the request to only return documents of the given doctype using `custom_doctype` field (see [mapping.rb](https://github.com/data-axle/elastic_record/blob/es6/lib/elastic_record/index/mapping.rb#L19)).

Right now, the expectation is that implementations of the library should manually set the `custom_doctype` value in query/filter requests. Though that isn't very intuitive, so an automatic solution that behaves similar to the current release of this library would be better. I found that to be difficult though since there are a lot of places, especially [documents.rb](https://github.com/data-axle/elastic_record/blob/es6/lib/elastic_record/index/documents.rb), that would need some conditional injections to the body.

Reply:
How does this work with code that is passing in a doctype currently? The old index and the new won't be in the same place.

Reply:
Hmm, yeah, it doesn't work for that case :( Sigh, complications of trying to upgrade a library with backwards compatibility between major versions.

I'm going to have to brainstorm this...this multi-version support is throwing me for a loop.

Reply:
I think it's better to approach it from a "does the index have multiple doc types" versus "what version of ES is it".